### PR TITLE
Using Cutter (r2) instead of IDA for symbolic execution section

### PR DESCRIPTION
**Pull request by cpholguera**
As part of the restructuring of chapter 0x5c we've noticed that there was a sentence telling that you could use IDA Pro Freeware for reversing Android Native Code. However it does not support ARM. Looking further in the chapter I've found that in the ### Binary Analysis Frameworks section IDA is being used so I switched to Cutter. This way everyone has the possibility to try out and follow our examples. 

I've put this on a separate PR because #1332 is already big enough. That PR actually splits "### Binary Analysis Frameworks" but I will take care of fixing that on my side once this is merged.

Also I will write a ticket on introducing Cutter that will be available once #1332 is merged.

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit 341b600d4b21f75ce49603978ac0d81801647bc4: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 4d52462a4e71dd6215a7d3be4f83513886171e63: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05c-Reverse-Engineering-and-Tampering.md [hunk 1]
```diff
@@ -1186,123 +1186,127 @@ $ adb shell /data/local/tmp/validate 12345
...
+
+If you look in the graph view you can see a loop with some XOR-magic happening at 0x00001784, which supposedly decodes the input string.
+
+![Loop](Images/Chapters/0x05c/loop_1784.png)
+
+Starting from 0x000017dc, you can see a series of decoded values compared with values from further subfunction calls.
+
+![Decoded values being compared](Images/Chapters/0x05c/values_compare_17dc.png)
+
+Even though this doesn't look like highly sophisticated stuff, you'd still need to analyze more to completely reverse this check and generate a license key that passes it. Now comes the twist: dynamic symbolic execution enables you to construct a valid key automatically! The symbolic execution engine maps a path between the first instruction of the license check (0x00001760) and the code that prints the "Product activation passed" message (0x00001840) to determine the constraints on each byte of the input string.
```
Comment (sushi2k) [inline Document/0x05c-Reverse-Engineering-and-Tampering.md]: Even though this doesn't look like highly sophisticated stuff,  -> Even though this doesn't look highly sophisticated,

Comment (cpholguera) [reply Document/0x05c-Reverse-Engineering-and-Tampering.md]: Done, thanks!

**Comment by OMTGreleaser** [comment]
Results for commit 21679e3ccd47b9a99c02c2ec543e06f364b4fa98: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Pull request merged**